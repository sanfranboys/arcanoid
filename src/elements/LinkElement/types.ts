export type LinkElementTypes ={
  link:string
  className?:string
  onClick?:(e?:Event)=>void
}
