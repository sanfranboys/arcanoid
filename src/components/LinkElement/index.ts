export {default as LinkElement} from './LinkElement'
