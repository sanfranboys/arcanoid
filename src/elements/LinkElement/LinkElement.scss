.link-component {
  color: var(--link);
  text-decoration: none;
  cursor: pointer;
}

.link-component:hover {
  color: var(--link-hover);
}
