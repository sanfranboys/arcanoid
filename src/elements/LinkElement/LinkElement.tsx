import React, { FC } from 'react'
import { Link } from 'react-router-dom'
import classNames from '@/utils/'
import { LinkElementTypes } from './types'
import './LinkElement.scss'

const LinkElement: FC<LinkElementTypes> = ({ className, link, children }) => (
  <Link to={link} className={classNames('link-component', className)}>
    {children}
  </Link>
)
export default LinkElement
