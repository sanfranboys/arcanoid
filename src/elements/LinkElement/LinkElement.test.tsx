import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { create } from 'react-test-renderer'
import LinkElement from './LinkElement'

describe('LinkElement testing', () => {
  const props = {
    link: '/link',
    className: 'mix-className',
    children: 'Link text here',
  }

  describe('LinkElement snapshot testing', () => {
    it('snapshot matches', async () => {
      const tree = create(
        <BrowserRouter>
          <LinkElement {...props} />
        </BrowserRouter>
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
