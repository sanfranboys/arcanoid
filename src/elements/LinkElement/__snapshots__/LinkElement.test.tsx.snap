// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkElement testing LinkElement snapshot testing snapshot matches 1`] = `
<a
  className="link-component mix-className"
  href="/link"
  onClick={[Function]}
>
  Link text here
</a>
`;
